src/Command/CreateClassCacheCommand.php

```diff
@@ -30,8 +30,26 @@
  *
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
-class CreateClassCacheCommand extends ContainerAwareCommand
+class CreateClassCacheCommand extends Command
 {
+    /**
+     * @var string
+     */
+    private $cacheDir;
+
+    /**
+     * @var bool
+     */
+    private $debug;
+
+    public function __construct(string $cacheDir, bool $debug)
```

Comment:
Maybe include a first, nullable `$name` argument: https://github.com/symfony/symfony/blob/82d5ff075a4f679bbf874f201e4351855deb67d6/src/Symfony/Component/Console/Command/Command.php#L69, and make both arguments optional, with deprecations in case these two arguments are not provided.

It is a bit more backwards-compatible.

Reply:
How do you propose to provide the default value for these arguments? IIUC, the use case which we are trying to cover with this BC bridge is when this class is explicitly instantiated (or extended) by the user. AFAIK, with this scenario we can't use conditional injection via compiler pass or `func_get_arg()` with a ghost container argument.

Reply:
Oh right the example I provided does not apply since I did not stop extending the deprecated class, that won't work indeed.